sources/models/src/modeled_types/kubernetes.rs

```diff
@@ -437,3 +437,120 @@ mod test_kubernetes_bootstrap_token {
         }
     }
 }
+
+// =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=   =^..^=
+
+/// EvictionHardKey represents a string that contains a valid Kubernetes eviction hard
+/// signal. There are few valid eviction hard signals [memory.available], [nodefs.available],
+/// [imagefs.available], and [nodefs.inodesFree].
+/// https://kubernetes.io/docs/tasks/administer-cluster/out-of-resource/
+
+
+#[derive(Debug, Clone, Eq, PartialEq, Hash)]
+pub struct EvictionHardKey {
+    inner: String,
+}
+
+impl TryFrom<&str> for EvictionHardKey {
+    type Error = error::Error;
+
+    fn try_from(input: &str) -> Result<Self, Self::Error> {
+        let evitionsignal = vec![
```

Comment:
we can also use enum here for example:  https://github.com/bottlerocket-os/bottlerocket/blob/develop/sources/models/src/modeled_types/ecs.rs#L189
However, I don't know which aligns more to our existing style, I would let other comment on what we should use.

Reply:
I was trying to use enum to handle this part; however, I think enum could be not that efficient and clear there because too many works on dealing with variables which contains dots like `memory.available`.  I got some ideas that if we can use rename attributes for each enum variant `#[serde(rename = "memory.available")] `, and I'm not sure if that worths. Appreciate for the comments! :)

Reply:
Another plus to just having it as an enum is being able to reuse it elsewhere for matching/validation etc.

I also wonder if we even need any of the additional traits here for converting to/from strings if `EvictionHardKey` itself was the enum. We've never tried that before but I don't see a clear reason why we can't just rely on serde's deserialization trait for validation.

Reply:
@gthao313 I think the rename attributes you mentioned above _should_ allow us to use an enum.  If so, that's probably the preferred route as other folks mentioned.